core/src/queue.rs

```diff
@@ -53,25 +52,21 @@ impl AcceptedTransaction {
 
 fn check_signature_condition(
     account: &Account,
-    signatories: impl IntoIterator<Item = PublicKey>,
-) -> EvaluatesTo<bool> {
-    let where_expr = Where::new(EvaluatesTo::new_evaluates_to_value(
-        *account.signature_check_condition.0.expression.clone(),
-    ))
-    .with_value(
-        iroha_data_model::account::ACCOUNT_SIGNATORIES_VALUE
-            .parse()
-            .expect("ACCOUNT_SIGNATORIES_VALUE should be valid."),
-        account.signatories.iter().cloned().collect::<Vec<_>>(),
-    )
-    .with_value(
-        iroha_data_model::account::TRANSACTION_SIGNATORIES_VALUE
-            .parse()
-            .expect("TRANSACTION_SIGNATORIES_VALUE should be valid."),
-        signatories.into_iter().collect::<Vec<_>>(),
-    );
-
-    EvaluatesTo::new_unchecked(where_expr)
+    transaction_signatories: impl IntoIterator<Item = PublicKey>,
+) -> bool {
+    let transaction_signatories = transaction_signatories.into_iter().collect::<Vec<_>>();
+    let account_signatories = &account.signatories;
+    match &account.signature_check_condition {
+        SignatureCheckCondition::AnyAccountSignatures => account_signatories
+            .iter()
+            .any(|transaction_signatory| transaction_signatories.contains(transaction_signatory)),
+        SignatureCheckCondition::AllAccountSignatures => account_signatories
+            .iter()
+            .all(|transaction_signatory| transaction_signatories.contains(transaction_signatory)),
+        SignatureCheckCondition::AllCustomSignatures(required_signatures) => required_signatures
+            .iter()
+            .all(|required_signature| transaction_signatories.contains(required_signature)),
+    }
```

Comment:
Maybe this logic should be method of `SignatureCheckCondition` itself?

Reply:
Done